src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ClientService.java

```diff
@@ -418,6 +416,23 @@ public ClientType deleteTlsCertificate(ClientId id, String certificateHash)
         return mergeClientListsDistinctively(globalClients, localClients);
     }
 
+    /**
+     * Registers a client
+     * @param clientId client to register
+     * @throws GlobalConfOutdatedException
+     * @throws ClientNotFoundException
+     */
+    public void registerClient(ClientId clientId) throws GlobalConfOutdatedException, ClientNotFoundException {
+        ClientType client = getLocalClientOrThrowNotFound(clientId); // throws if client doesn't exist
+        try {
+            managementRequestSenderService.sendClientRegisterRequest(clientId);
+            client.setClientStatus(ClientType.STATUS_REGINPROG);
+            clientRepository.saveOrUpdate(client);
```

Comment:
Old implementation also has "cannot register owner" check that we are missing at the moment: https://github.com/nordic-institute/X-Road/blob/19578462600ce3ec1832e3af5ff2b0f1b1cfe01c/src/proxy-ui/app/controllers/clients_controller.rb#L234-L236

Should we have that? Or was it intentionally left out (for init, maybe?). 

I do not yet have clear vision of how init will eventually work, maybe one option would be to include "cannot register owner" check now, and revise that later if needed, when we implement init?

Reply:
I added a check for this. It should be changed when implementing the init functionality – if needed.